### PR TITLE
Remove use of AssemblyLoadContext

### PR DESCRIPTION
**Pull request by rynowak**
This will be needed for the switch back to net46.

### CODE REVIEW

**Comment by rynowak** [comment]
I can't build right now, so I just did what I thought seemed right. Will verify before merging.

File: src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs [hunk 1]
```diff
@@ -185,7 +185,7 @@ private static bool IsError(Diagnostic diagnostic)
 
         public static Assembly LoadAssembly(MemoryStream assemblyStream, MemoryStream pdbStream)
         {
-            var assembly = AssemblyLoadContext.Default.LoadFromStream(assemblyStream, pdbStream);
+            var assembly = Assembly.Load(assemblyStream.GetBuffer(), pdbStream.GetBuffer());
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Mvc.Razor/Internal/DefaultRoslynCompilationService.cs]: GetBuffer isn't right. That returns the entire underlying array even if it has more than what was written. You need to pass ToArray here.

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**